charon/src/ast/expressions.rs

```diff
@@ -451,13 +465,20 @@ pub struct ConstantExpr {
     Debug, Clone, EnumToGetters, EnumAsGetters, EnumIsA, Serialize, Deserialize, Drive, DriveMut,
 )]
 pub enum Rvalue {
+    /// Returns the operand unchanged.
```

Comment:
I think the real distinction is rather than operands can be used in different situations (for instance as inputs to functions), so this simply lifts operands to rvalues. By contrast, a borrow can not be used directly as input to a function (we have to create the borrow, store it in a temporary value, then move it).